packages/fx-core/templates/core/v3Migration/upgradeReport.md

```diff
@@ -1,50 +1,49 @@
-# Upgrade Summary
+# Review project upgrade and changes
 
-Congratulations! You can continue your development now. Your project has been upgraded to work with the latest Teams Toolkit.
+Your project was automatically upgraded to work with this version of Teams Toolkit. The upgrade is one-way, and you will not be able to use the version of Teams Toolkit in which it was originally created unless you [roll back the changes](#how-to-roll-back-after-the-upgrade).
 
-Most of existing features behave similar with previous Teams Toolkit. You can visit https://aka.ms/teamsfx-v5.0-guide to learn the new features. And read [Important changes to your development flow](#important-changes-to-your-development-flow) to understand the changes to your development flow.
-
-Visit https://aka.ms/teams-toolkit-5.0-upgrade to learn more about the upgrade.
+> If you encountered a problem after the automatic upgrade, review the [known issues](https://aka.ms/teams-toolkit-5.0-upgrade#known-issues). If you are still having trouble, contact ttkfeedback@microsoft.com or [file an issue on GitHub](https://github.com/officedev/teamsfx/issues) with details about the problem.
 
 ## Important changes to your development flow
 
-There's no immediate action required from you. This part illustrates what's changed to your development flow.
-> If you encountered any errors or issues after upgrade, you could try the following steps or visit [known issues](https://aka.ms/teams-toolkit-5.0-upgrade#known-issues) for details.
+The new features in this release make developing your apps simpler and more flexible, but also include changes to how the toolkit creates projects, automates configuration, and deploys resources. Visit [the upgrade guide](https://aka.ms/teamsfx-v5.0-guide) for more info about all of the changes.
+
+Some of the changes you may immediately notice are:
+
+* Configuration of the lifecycle management using Provision, Deploy, and Publish are now fully customizable and expressed in `teamsapp.local.yml` and `teamsapp.yml`. [More info](https://aka.ms/teamsfx-v5.0-guide#project-files)
+* Configuration and values that were saved in `.fx/config` and `.fx/state` are now handled with environment files and saved to `/env` by default. [More info](https://aka.ms/teams-toolkit-5.0-upgrade#environment-management)
+* The changes to use environment files give greater flexibility of configuring which resources are used to provision, but may require some manual steps when creating new environments. [More info](https://aka.ms/teams-toolkit-5.0-upgrade#environment-management)
+* Teams Toolkit no longer prompts you for which environment to use when using the Launch targets in `launch.json`. You can edit `.vscode/launch.json` and configure which environments are used. [More info](https://aka.ms/teams-toolkit-5.0-upgrade#launch-your-app)
```

Comment:
Confirmed with @kuojianlu , this limitation no longer exists in V5 stable release. Teams Toolkit V5 will ask users select an env like Teams Toolkit V4. Can you help remove this?

Reply:
Updated in the new PR: https://github.com/OfficeDev/TeamsFx/pull/8710